[wasm] Emscripten 2.0.34 bump

radekdoulik opened the pull request:
Performance wise it looks similar to 2.0.23. Didn't notice any regression here. The managed startup is slightly slower, in second run I measured 223ms, so it is probably just measurement error.

| measurement | 2.0.23 time | 2.0.34 |
|-:|-:|-:|
|                    AppStart, Page show |    63.0000ms |    63.4458ms |
|                AppStart, Reach managed |   222.5600ms |   226.1304ms |
|        Exceptions, NoExceptionHandling |     0.0921us |     0.0843us |
|                   Exceptions, TryCatch |     0.0955us |     0.0885us |
|              Exceptions, TryCatchThrow |     0.0026ms |     0.0026ms |
|             Exceptions, TryCatchFilter |     0.1018us |     0.0929us |
|       Exceptions, TryCatchFilterInline |     0.0869us |     0.0800us |
|        Exceptions, TryCatchFilterThrow |     0.0033ms |     0.0034ms |
| Exceptions, TryCatchFilterThrowApplies |     0.0025ms |     0.0025ms |
|         Json, non-ASCII text serialize |     9.0376ms |     7.9638ms |
|       Json, non-ASCII text deserialize |    14.0856ms |    11.7460ms |
|                  Json, small serialize |     0.2351ms |     0.2203ms |
|                Json, small deserialize |     0.3556ms |     0.3359ms |
|                  Json, large serialize |    69.6667ms |    63.6420ms |
|                Json, large deserialize |    99.6792ms |    92.1754ms |
|              WebSocket, PartialSend 1B |     0.0017ms |     0.0017ms |
|            WebSocket, PartialSend 64KB |     0.0653ms |     0.0658ms |
|             WebSocket, PartialSend 1MB |     0.9636ms |     0.9364ms |
|           WebSocket, PartialReceive 1B |     0.0026ms |     0.0025ms |
|         WebSocket, PartialReceive 10KB |     0.0060ms |     0.0040ms |
|        WebSocket, PartialReceive 100KB |     0.0000us |     0.0000us |

radekdoulik committed:
Initial changes for emscripten 2.0.34

radekdoulik committed:
Use emcc-link.rsp in build targets

radekdoulik committed:
Use updated docker images

vargaz commented:
Might need this as well:
      <_EmccCommonFlags Include="-s ENVIRONMENT=&quot;web,webview,worker,node,shell&quot;" />

Newer versions of emscripten no longer build with 'shell' support by default, i.e. running under v8.

vargaz commented:
Fix for the build failure:
https://gist.github.com/vargaz/4bfce768ed19d283ddc31be124cc82a9

Commit:
Fix compiler warnings

Co-authored-by: Zoltan Varga <vargaz@gmail.com>

radekdoulik committed:
Put `--profiling-funcs` to `_EmccLinkFlags`

radekdoulik committed:
Fix build

    src/mono/mono/mini/mini-runtime.c:3407:25: error: ‘invoke’ undeclared (first use in this function); did you mean ‘revoke’?
       3407 |                         invoke = mono_marshal_get_runtime_invoke_dynamic ();

radekdoulik committed:
Add shell to the environment

Environment setting https://github.com/emscripten-core/emscripten/blob/2.0.34/src/settings.js#L616-L641

From emscripten 2.0.25 release notes

    - Support for the 'shell' environment is now disabled by default.  Running under
      `d8`, `js`, or `jsc` is not something that most emscripten users ever want to
      do, so including the support code is, more often than not, unnecessary.  Users
      who want shell support can enable it by including 'shell' in `-s ENVIRONMENT`
      (#14535).

Example of the the size increase for bench sample:

    -a---          12/10/2021  3:35 PM         382113 dotnet.js
    -a---          12/13/2021 10:37 AM         383589 dotnet.js

radekdoulik commented:
Related: https://github.com/dotnet/runtime/issues/62708

radekdoulik committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Add emcc-link.rsp to PlatformManifestFileEntry

pavelsavara commented on src/mono/wasm/wasm.proj:
```diff
@@ -64,21 +66,23 @@
-      <_EmccCommonFlags Include="-s STRICT_JS=1" />      
-      <_EmccCommonFlags Include="-s EXPORT_NAME=&quot;'createDotnetRuntime'&quot;" />
-      <_EmccCommonFlags Include="-s MODULARIZE=1"/>
       <_EmccCommonFlags Include="-s EXPORT_ES6=1" Condition="'$(WasmEnableES6)' == 'true'" />
```
EXPORT_ES6 is probably also link flag, right ?

radekdoulik committed:
Feedback

https://github.com/emscripten-core/emscripten/blob/2fda25eea756c78c8cb024aa5b6c2b188bf7990f/src/settings.js#L1173-L1176

-s EXPORT_ES6 is link option

radekdoulik committed:
Remove unused variable

radekdoulik committed:
Use new Emscripten workload

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

lewing commented:
/azp run runtime-extra-platforms

lewing committed:
Add the emcc-link.rsp

lewing commented:
looks like we need to bump Emscripten in icu https://github.com/dotnet/icu/pull/177

radekdoulik committed:
Bump dotnet icu

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

lewing commented:
/azp run runtime-extra-platforms

SamMonoRT commented:
@naricc - any corresponding change needed anywhere in BDN or Perf repo ?

lewing committed:
Merge branch 'main' into pr-emscripten-34-bump

lewing commented:
/azp run runtime-extra-platforms

lewing commented:
/azp run runtime runtime-extra-platforms

lewing commented:
/azp run runtime

lewing commented:
/azp run runtime-extra-platforms

radical committed:
Merge remote-tracking branch 'origin/main' into pr-emscripten-34-bump

radical commented:
/azp run runtime-extra-platforms

lewing commented:
runtime (Libraries Build windows net48 x86 Release) failure is https://github.com/dotnet/runtime/issues/64153

BrzVlad commented on src/mono/mono/mini/mini-generic-sharing.c:
```diff
@@ -1763,7 +1765,10 @@ mini_get_interp_in_wrapper (MonoMethodSignature *sig)
-	return_native_struct = sig->ret->type == MONO_TYPE_VALUETYPE && sig->pinvoke && !sig->marshalling_disabled;
+#ifndef DISABLE_JIT
+	return_native_struct =
+#endif
```
why endif here and not whole statement ?

radekdoulik committed:
Feedback

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

radekdoulik commented:
/azp run runtime-extra-platforms

radekdoulik committed:
Do not add linker flags to compilation

radekdoulik commented:
/azp run runtime-extra-platforms

radekdoulik committed:
Split `_EmccLDFlags`

To avoid these errors:

WasmApp.Native.targets(342,5): error : Failed to compile .../Microsoft.CodeAnalysis.CSharp.dll.bc -> /datadisks/disk1/work/B9F209B7/w/B1710A2F/e/wasm_build/obj/wasm/for-build/Microsoft.CodeAnalysis.CSharp.dll.o
WasmApp.Native.targets(342,5): error : emcc: warning: linker setting ignored during compilation: 'TOTAL_MEMORY' [-Wunused-command-line-argument]
WasmApp.Native.targets(342,5): error : emcc: warning: linker setting ignored during compilation: 'ERROR_ON_UNDEFINED_SYMBOLS' [-Wunused-command-line-argument]

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-emscripten-34-bump

radekdoulik committed:
Bump emscripten workload

radekdoulik commented:
/azp run runtime-extra-platforms

Commit:
Update windows helix queues

radekdoulik commented:
/azp run runtime-extra-platforms

lewing committed:
Merge branch 'main' into pr-emscripten-34-bump

radical committed:
[wasm] Fix WBT running with nodejs

When running with nodejs, the managed app would exit with code 42, but
node would exit with 1. Use `process.exit` for node, instead of
`mono_wasm_exit`.

radical commented:
/azp run runtime-extra-platforms

radical commented:
Failures:

- `Microsoft.Extensions.Logging.Generators.Roslyn3.11.Tests` - timing out, likely OOM
- https://github.com/dotnet/runtime/issues/64724
- https://github.com/dotnet/runtime/issues/64725
- https://github.com/dotnet/runtime/issues/64727

These will be investigated after this PR is merged.

sbomer commented:
Why was this merged with known runtime-extra-platforms failures? I ask because we are trying to keep get runtime and runtime-extra-platforms to green to make it easier to detect new failures. Would it be reasonable to disable the failing tests before merging in the future? Just curious what we can do to meet our build health goals without unnecessarily slowing down development.

radical commented:
> Why was this merged with known runtime-extra-platforms failures?

This was part of a bigger mess, which started with an update dependencies PR from emsdk getting merged, which bumped the emscripten version being used, which broken Wasm.Build.Tests (on `main`, and PRs), which got mixed with many helix queues getting removed, and all that combining to cause a badly red `main`, and lots of PRs being blocked. Merging this unblocked them, and made `main` much less red. And I opened a PR to disable those tests today (https://github.com/dotnet/runtime/pull/64759).

> I ask because we are trying to keep get runtime and runtime-extra-platforms to green to make it easier to detect new failures. Would it be reasonable to disable the failing tests before merging in the future? Just curious what we can do to meet our build health goals without unnecessarily slowing down development.

I completely agree with that, and avoid merging with any failures caused by the PR. But this was an unusual case.

sbomer commented:
Thanks @radical, I appreciate the explanation
